core/src/main/scala/api/attributes.scala

```diff
@@ -0,0 +1,40 @@
+package sfs
+package api
+
+object attributes {
+
+  final case class Permissions(
+    ownerRead: Boolean, ownerWrite: Boolean, ownerExecute: Boolean,
+    groupRead: Boolean, groupWrite: Boolean, groupExecute: Boolean,
+    otherRead: Boolean, otherWrite: Boolean, otherExecute: Boolean
+  )
+  object Permissions {
+    implicit def empty: api.Empty[Permissions] =
+      api.Empty(Permissions(false, false, false, false, false, false, false, false, false))
+  }
```

Comment:
I don't care if it's changed at this second, but I am a big disbeliever in the "nine booleans" approach to this sort of thing and would like this to go away eventually. The set of possible permissions is vastly larger than these nine bits - these are, specifically, the unix filesystem interface permissions, and as a result there's little reason not to store it just as the unix filesystem does, which is to say as an integer of some width. Not sure if I already shared this, but here's one from one of my fuse projects. This includes the translation between java and fuse.
```scala
final case class UnixAttrs(perms: UnixPerms, times: UnixTimes, owners: UnixOwners)
final case class UnixOwners(uid: Int, gid: Int)
final case class UnixTimes(created: FileTime, modified: FileTime, accessed: FileTime)
final class UnixPerms private (val mask: Long) {
  def bits: Set[Long]                                = FuseBitsSet filter (bit => (bit & mask) != 0)
  def java: Set[PosixFilePermission]                 = bits map FuseToJava
  def fuse: ModeWrapper                              = new ModeWrapper(mask)
  def attr: FileAttribute[jSet[PosixFilePermission]] = PosixFilePermissions asFileAttribute java.asJava

  def isExecutable = java(OTHERS_EXECUTE) | java(GROUP_EXECUTE) | java(OWNER_EXECUTE)

  override def toString = permString(mask)
}

object UnixPerms {
  private lazy val Part      = "[-r][-w][-x]"
  private lazy val PermRegex = ("^" + (Part * 3) + "$").r
  private lazy val JavaBits  = Vector[PosixFilePermission](
    OWNER_READ,
    OWNER_WRITE,
    OWNER_EXECUTE,
    GROUP_READ,
    GROUP_WRITE,
    GROUP_EXECUTE,
    OTHERS_READ,
    OTHERS_WRITE,
    OTHERS_EXECUTE
  )
  private lazy val FuseBits = Vector[Long](
    S_IRUSR,
    S_IWUSR,
    S_IXUSR,
    S_IRGRP,
    S_IWGRP,
    S_IXGRP,
    S_IROTH,
    S_IWOTH,
    S_IXOTH
  )
  private lazy val FuseBitsSet: Set[Long] = FuseBits.toSet
  private lazy val Letters: Vector[Char]  = "rwxrwxrwx".toVector
  private lazy val JavaToFuse             = JavaBits zip FuseBits toMap
  private lazy val FuseToJava             = FuseBits zip JavaBits toMap

  private def fuseMask(perms: String): Long =
    ( for ((perm, ch) <- FuseBits zip perms.toSeq ; if ch != '-') yield perm ).foldLeft(0L)(_ | _)

  private def permString(mask: Long): String =
    ( for ((perm, ch) <- FuseBits zip Letters) yield if ((mask & perm) == 0) '-' else ch ) mkString ""

  private def fuseMask(path: Path): Long =
    ( path.direct.permissions map JavaToFuse ).foldLeft(0L)(_ | _)

  def apply(perms: String): UnixPerms = perms match {
    case PermRegex() => apply(fuseMask(perms))
    case _           => sys.error(s"Invalid permission string: $perms")
  }

  def apply(bits: Long): UnixPerms     = new UnixPerms(bits)
  def apply(m: ModeWrapper): UnixPerms = apply(m.mode)
  def apply(p: Path): UnixPerms        = apply(fuseMask(p))
  def apply(f: File): UnixPerms        = apply(f.toPath)

  def `777` = apply("rwxrwxrwx")
  def `755` = apply("rwxr-xr-x")
  def `644` = apply("rw-r--r--")
  def `444` = apply("r--r--r--")
  def `555` = apply("r-xr-xr-x")
}
```